test/util/schemaform-utils.jsx

```diff
@@ -39,8 +43,11 @@ export class DefinitionTester extends React.Component {
   handleChange = (data) => {
     const state = this.props.state;
     const uiSchema = this.state.uiSchema;
-    const formData = _.merge(this.props.formData, data);
-    let schema = updateRequiredFields(this.state.schema, uiSchema, formData);
+    const formData = this.state.schema.type === 'array'
+      ? data
+      : _.merge(this.props.formData, data);
+    let schema = updateItemsSchema(this.state.schema, formData);
+    schema = updateRequiredFields(schema, uiSchema, formData);
```

Comment:
Is there an argument against making a little wrapper function for calling `updateItemsSchema() and `updateRequiredFields`? The latter seems to always be preceded by the former, so I'd hate for some modification down the road to only call one or the other, y'know? Debugging that would be such a pain!

Heh, on the other hand, we seem to be running out of names for these functions and still have them make sense!